components/feature/push/src/main/java/mozilla/components/feature/push/Connection.kt

```diff
@@ -66,6 +82,30 @@ interface PushConnection : Closeable {
     fun isInitialized(): Boolean
 }
 
+/**
+ * An implementation of [PushConnection] for the native component using the [PushAPI].
+ *
+ * Implementation notes: There are few important concepts to note here - for WebPush, the identifier that is required
+ * for notifying clients is called a "scope". This can be a site's host URL or an other uniquely identifying string to
+ * create a push subscription for that given scope.
+ *
+ * In the native Rust component the concept is similar, in that we have a unique ID for each subscription, however
```

Comment:
The unique thing here is the pair `(uaid, channelId)`, right? Which is hidden from consumers of this API. E.g. this code running on different devices, when encountering the same `scope`, won't collapse into the same notification endpoint on the autopush side.

Reply:
The `uaid` isn't something that's exposed to us in anyway so it's hidden to a-c as well, but I see the point you're making, that we should call out the uniqueness is per device as well.